### PR TITLE
Update integration-tests README

### PR DESCRIPTION
**Pull request by chudilka1**
[BCF-3396](https://smartcontract-it.atlassian.net/browse/BCF-3396)

Audit and update integration-tests READMEs.

[BCF-3396]: https://smartcontract-it.atlassian.net/browse/BCF-3396?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by EasterTheBunny — DISMISSED** [review]

File: .github/E2E_TESTS_ON_GITHUB_CI.md [hunk 1]
```diff
@@ -1,58 +1,87 @@
...
 
-These workflows are designed to run automatically at crucial stages of the software development process, such as on every commit in a PR, nightly or before release.
+## Scheduled test workflows
+
+These workflows are designed to run on every commit in a PR, nightly or before release (see `triggers` in [e2e-tests.yaml](./e2e-tests.yml)).
 
 ### PR E2E Tests
 
-Run on every commit in a PR to ensure changes do not introduce regressions.
+Smoke tests triggered on every commit in a PR to ensure changes do not introduce regressions.
```
Comment (lukaszcl) [inline .github/E2E_TESTS_ON_GITHUB_CI.md]: @chudilka1 these are not only smoke tests. In fact, we run almost all tests we have except soak or performance tests.

File: .github/E2E_TESTS_ON_GITHUB_CI.md [hunk 2]
```diff
@@ -1,58 +1,87 @@
...
+**Workflow:** [nightly-e2e-tests.yml](https://github.com/smartcontractkit/chainlink/blob/develop/.github/workflows/run-nightly-e2e-tests.yml)
 
 ### Release E2E Tests
 
-This section contains automatic workflows triggering E2E tests at release.
+E2E tests triggered on a release tag.
+
+#### Smoke Tests
+
+**Workflow:** [integration-tests.yml](https://github.com/smartcontractkit/chainlink/blob/develop/.github/workflows/integration-tests.yml)
```
Comment (lukaszcl) [inline .github/E2E_TESTS_ON_GITHUB_CI.md:39]: @chudilka1 these are not only smoke tests. In fact, we run almost all tests we have except soak or performance tests.

File: .github/E2E_TESTS_ON_GITHUB_CI.md [hunk 3]
```diff
@@ -1,58 +1,87 @@
...
+1. Reference sources:
+   1. [Integration-Tests configurations](../integration-tests/testconfig/README.md);
+   2. [CTF Test Config](https://github.com/smartcontractkit/chainlink-testing-framework/blob/main/lib/config/README.md).
+2. Defaults and overrides should be stored (committed) in repository under `../integration-tests/testconfig/<product>/overrides/<override-name>.toml` (see example [here](../integration-tests/testconfig/ocr2/overrides/base_sepolia.toml)).
+3. Use `test_config_override_path` to point to an override config. For example: `test_config_override_path="testconfig/ocr2/overrides/base_sepolia.toml"`
 
-## Test Secrets
+#### Test Secrets
 
-For security reasons, test secrets and sensitive information are not stored directly within the test config TOML files. Instead, these secrets are securely injected into tests using environment variables. For a detailed explanation on managing test secrets, refer to our [Test Secrets documentation](https://github.com/smartcontractkit/chainlink-testing-framework/blob/main/lib/config/README.md#test-secrets).
```
Comment (lukaszcl) [inline .github/E2E_TESTS_ON_GITHUB_CI.md:52]: > For security reasons, test secrets and sensitive information are not stored directly within the test config TOML files. Instead, these secrets are securely injected into tests using environment variables.

@chudilka1 why you removed this?

File: .github/E2E_TESTS_ON_GITHUB_CI.md [hunk 4]
```diff
@@ -1,58 +1,87 @@
...
+> Pay attention to never store/expose/commit your test secrets in repository.
 
-If you need to run a GitHub workflow using custom secrets, please refer to the [guide on running GitHub workflows with your test secrets](https://github.com/smartcontractkit/chainlink-testing-framework/blob/main/lib/config/README.md#run-github-workflow-with-your-test-secrets).
+Test secrets allow provisioning and override the sensitive data such as EOA's private key, RPCs, Docker registry links, etc.
 
-## About the E2E Test Reusable Workflow
+Reference sources:
 
-For information on the E2E Test Reusable Workflow, visit the documentation in the [smartcontractkit/.github repository](https://github.com/smartcontractkit/.github/blob/main/.github/workflows/README.md).
+1. [BASE64_CONFIG_OVERRIDE](../integration-tests/testconfig/README.md#base64_config_override).
```
Comment (lukaszcl) [inline .github/E2E_TESTS_ON_GITHUB_CI.md:88]: @chudilka1 I'd explicitly say that **test configs** (link to docs in CTF) are used for non-secrets and **test secrets** for secrets (link to docs in CTF)

**Review by lukaszcl — COMMENTED** [review]
@chudilka1 it lgtm, but I've added some comments.

File: .github/E2E_TESTS_ON_GITHUB_CI.md (hunk 1 above)
Comment (anirudhwarrier) [reply .github/E2E_TESTS_ON_GITHUB_CI.md]: @lukaszcl looking at https://github.com/smartcontractkit/chainlink/blob/develop/.github/e2e-tests.yml every single test that is in either `PR E2E Core Tests` or `Merge Queue E2E Core Tests` are all smoke tests. The only test that is not a smoke is `migration/upgrade_version_test.go`, but even that is more or less like a smoke, but just checks upgrade.

File: .github/E2E_TESTS_ON_GITHUB_CI.md (hunk 3 above)
Comment (chudilka1) [reply .github/E2E_TESTS_ON_GITHUB_CI.md:52]: I've removed it, as there are guidelines and references to the corresponding documents that clearly explain that. Moreover, tests will fail if there are no test secrets. There is no need to duplicate this information everywhere.

File: .github/E2E_TESTS_ON_GITHUB_CI.md (hunk 1 above)
Comment (chudilka1) [reply .github/E2E_TESTS_ON_GITHUB_CI.md]: Updated, removed - Smoke

File: .github/E2E_TESTS_ON_GITHUB_CI.md (hunk 2 above)
Comment (chudilka1) [reply .github/E2E_TESTS_ON_GITHUB_CI.md:39]: Updated to "Integration (smoke) Tests"

File: .github/E2E_TESTS_ON_GITHUB_CI.md (hunk 4 above)
Comment (chudilka1) [reply .github/E2E_TESTS_ON_GITHUB_CI.md:88]: I've added the risk sections to address it. Though, IMHO, there is no need, as there is a section about [test/node configuration](https://github.com/smartcontractkit/chainlink/pull/14577/files#diff-c8e584763fd2bd4e5ef372d8f5221ed841afbb22920e64af239148786d6caa69R61).

File: .github/E2E_TESTS_ON_GITHUB_CI.md [hunk 5]
```diff
@@ -1,58 +1,90 @@
...
 
-E2E tests utilize TOML files to define their parameters. Each test is equipped with a default TOML config, which can be overridden by specifying an alternative TOML config. This allows for running tests with varied parameters, such as on a non-default blockchain network. For tests executed on GitHub CI, both the default configs and any override configs must reside within the git repository. The `test_config_override_path` workflow input is used to provide a path to an override config.
+> [!TIP]
+> Use `gh` CLI commands to run workflows from local machine.
 
-Config overrides should be stored in `testconfig/*/overrides/*.toml`. Placing files here will not trigger a rebuild of the test runner image.
+#### Configuration Overrides
 
-**Important Note:** The use of `base64Config` input is deprecated in favor of `test_config_override_path`. For more details, refer to [the decision log](https://smartcontract-it.atlassian.net/wiki/spaces/TT/pages/927596563/Storing+All+Test+Configs+In+Git).
+> [!RISK]
```
Comment (AKhozya) [inline .github/E2E_TESTS_ON_GITHUB_CI.md]: Screenshot 2024-10-04 at 14 31 21
probably some other icon should be used, currently it's rendered as !RISK in "reading mode"

File: .github/E2E_TESTS_ON_GITHUB_CI.md [hunk 6]
```diff
@@ -1,58 +1,90 @@
...
+   1. [Integration-Tests configurations](../integration-tests/testconfig/README.md);
+   2. [CTF Test Config](https://github.com/smartcontractkit/chainlink-testing-framework/blob/main/lib/config/README.md).
+2. Defaults and overrides should be stored (committed) in repository under `../integration-tests/testconfig/<product>/overrides/<override-name>.toml` (see example [here](../integration-tests/testconfig/ocr2/overrides/base_sepolia.toml)).
+3. Use `test_config_override_path` to point to an override config. For example: `test_config_override_path="testconfig/ocr2/overrides/base_sepolia.toml"`
 
-## Test Secrets
+#### Test Secrets
 
-For security reasons, test secrets and sensitive information are not stored directly within the test config TOML files. Instead, these secrets are securely injected into tests using environment variables. For a detailed explanation on managing test secrets, refer to our [Test Secrets documentation](https://github.com/smartcontractkit/chainlink-testing-framework/blob/main/lib/config/README.md#test-secrets).
+> [!RISK]
```
Comment (AKhozya) [inline .github/E2E_TESTS_ON_GITHUB_CI.md]: Same as above, maybe formatting is not right?
Screenshot 2024-10-04 at 14 31 21

File: integration-tests/run-books/OCR.md [hunk 7]
```diff
@@ -0,0 +1,252 @@
...
+> These tests require logging in to Kubernetes cluster (`aws sso login`).
+> Do not use `-timeout` flag for Soak tests with Go command. It is set in `overrides.toml`
+
+Refer [Test config README](../testconfig/README.md) for more details about Soak tests configuration.
+
+##### Kubernetes
+
+Refer [COMMON COMMANDS#Kubernetes](#kubernetes). Override path as follows:
+
+With forwarders:
```
Comment (AKhozya) [inline integration-tests/run-books/OCR.md]: I'd add bullets for each case, it's hard to read this part. Optional though

File: integration-tests/testconfig/README.md [hunk 8]
```diff
@@ -1,136 +1,114 @@
...
+## Summary
+
+The `testconfig` package represents a storage of test configurations per product/service.
+
+> [!TIP]
+>
+> 1. The `testconfig.go` may `Save()` test configurations.
+> 2. To identify the configurations in use, run tests with the `TEST_LOG_LEVEL=debug`.
+
+## Configurations Files and Overrides Precedence
```
Comment (AKhozya) [inline integration-tests/testconfig/README.md:51]: Configurations => Configuration. Probably mistype

File: integration-tests/run-books/OCR.md [hunk 9]
```diff
@@ -0,0 +1,252 @@
...
+OR
+
+Run [On Demand OCR Soak Test](https://github.com/smartcontractkit/chainlink/actions/workflows/run-on-demand-ocr-soak-test.yml) workflow with `gh` as follows:
+
+```bash
+gh workflow run "on-demand-ocr-soak-test.yml" \
+--ref <branch> \
+-f chainlink_version="v<your.version>" \ # Optional, default is image created from develop branch. Not needed if you run tests against existing environment
+-f workflow_run_name="Any name" \ # Optional
+-f test_ids="<test_ID>" \ # see /chainlink/.github/e2e-tests.yml for IDS
```
Comment (joaoluisam) [inline integration-tests/run-books/OCR.md]: Just noticing that we mention test_ids but on example we use test to run. I do see a test id for ocr soak in e2e.yml.
Is this intentional or do we want to include both possibilities in examples?

File: integration-tests/testconfig/README.md [hunk 10]
```diff
@@ -185,168 +170,212 @@ To set custom per-chain config use `[NodeConfig.ChainConfigTOMLByChainID]`. Exam
...
+# Example 2: Setting `configure` to different values for the listed contracts will fail execution
+[OCR.Contracts.Settings."0xc1ce3815d6e7f3705265c2577F1342344752A5Eb"]
+configure = false
 
-## Contributing
+[OCR.Contracts.Settings."0x2f4FA21fCd917C448C160caafEC874032F404c08"]
+configure = true
 
-It's crucial to incorporate all new test configuration settings directly into the TOML configuration files, steering clear of using environment variables for this purpose. Our goal is to centralize all configuration details, including examples, within the same package. This approach simplifies the process of understanding the available configuration options and identifying the appropriate values to use for each setting.
```
Comment (AKhozya) [inline integration-tests/testconfig/README.md:337]: IMO I'd leave this to underline our strategy here

**Review by AKhozya — CHANGES_REQUESTED** [review]

**Review by joaoluisam — COMMENTED** [review]
Minor observation/question. 
Good job with this 🙌

File: .github/E2E_TESTS_ON_GITHUB_CI.md (hunk 5 above)
Comment (chudilka1) [reply .github/E2E_TESTS_ON_GITHUB_CI.md]: Thanks, changed to CAUTION, the invalid mark down.

File: .github/E2E_TESTS_ON_GITHUB_CI.md (hunk 6 above)
Comment (chudilka1) [reply .github/E2E_TESTS_ON_GITHUB_CI.md]: Updated to CAUTION, was an invlid md.

File: integration-tests/run-books/OCR.md (hunk 7 above)
Comment (chudilka1) [reply integration-tests/run-books/OCR.md]: Thanks, updated.

File: integration-tests/testconfig/README.md (hunk 8 above)
Comment (chudilka1) [reply integration-tests/testconfig/README.md:51]: Thanks, Updated

File: integration-tests/testconfig/README.md (hunk 10 above)
Comment (chudilka1) [reply integration-tests/testconfig/README.md:337]: Moved it to the very top of the doc (under [!IMPROTANT] alert), thanks

File: integration-tests/run-books/OCR.md (hunk 9 above)
Comment (chudilka1) [reply integration-tests/run-books/OCR.md]: Thanks, @joaoluisam , fixed.

**Review by AKhozya — APPROVED** [review]

**Review by joaoluisam — APPROVED** [review]

**Review by jmank88 — APPROVED** [review]

**Pull request merged**